pkg/pfconfig/conversion.go

```diff
@@ -0,0 +1,150 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pfconfig
+
+import (
+	"fmt"
+	"math"
+	"net"
+	"strconv"
+	"time"
+
+	"go.thethings.network/lorawan-stack/pkg/band"
+	"go.thethings.network/lorawan-stack/pkg/frequencyplans"
+	"go.thethings.network/lorawan-stack/pkg/ttnpb"
+)
+
+func formatFrequency(frequency uint64) string {
+	freq := float64(frequency) / 1000000
+	if freq*10 == math.Floor(freq*10) {
+		return fmt.Sprintf("%.1f", freq)
+	}
+	return fmt.Sprintf("%g", freq)
+}
+
+// Build builds a packet forwarder configuration for the given gateway, using the given frequency plan store.
+func Build(gateway *ttnpb.Gateway, store *frequencyplans.Store) (*Config, error) {
+	var c Config
+
+	host, portStr, err := net.SplitHostPort(gateway.GatewayServerAddress)
+	if err != nil {
+		host = gateway.GatewayServerAddress
+		portStr = "1700"
```

Comment:
Better to make this default also configurable if we decide to use a proxy later.

Reply:
This is only used if there is no port specified in `gateway.GatewayServerAddress`. `1700` is the de-facto standard port for UDP forwarders (which this package is targeting), so I think this is a sane fallback. When this package is actually used, the caller can make sure that a port is indicated in `gateway.GatewayServerAddress`.